src/plugins/data_source/server/legacy/configure_legacy_client.ts

```diff
@@ -96,15 +103,31 @@ const getQueryClient = async (
   config: DataSourcePluginConfigType,
   registeredSchema: any[],
   rootClient?: LegacyClient,
-  dataSourceId?: string
+  dataSourceId?: string,
+  request?: OpenSearchDashboardsRequest,
+  authRegistry?: IAuthenticationMethodRegistery
 ) => {
-  const {
+  let credential;
+  let {
     auth: { type },
-    endpoint: nodeUrl,
+    name,
   } = dataSourceAttr;
+  const { endpoint: nodeUrl } = dataSourceAttr;
```

Comment:
can we extract endpoint as before, eg. line 103

Reply:
`nodeUrl` is never reassigned. Therefore I used `const` instead

Reply:
Got it!